protocol-units/settlement/mcr/client/src/lib.rs

```diff
@@ -5,10 +5,10 @@ use tokio_stream::Stream;
 pub mod tests;
 
 #[cfg(feature = "mock")]
-mod mock;
+pub mod mock;
 
 #[cfg(feature = "mock")]
-pub use mock::*;
+pub use mock::McrSettlementClient;
```

Comment:
The re-export is still confusing: an import in some consumer code could look like

```rust
use mcr_settlement_client::McrSettlementClient;
```

It's not clear at all that it's a mock to a cursory reviewer.

This makes the intent much clearer:
```rust
use mcr_settlement_client::mock::McrSettlementClient;
```

Reply:
This reexport is the pattern use to switch from Eth and Mock feature. The mcr_settlement_client crate always expose a `McrSettlementClient` and the impl changes depending on the activated feature.
In the suzuka node, it's imported like that: `use mcr_settlement_client::McrSettlementClient;`
If we import using `use mcr_settlement_client::mock::McrSettlementClient;` we need to feature lock all these import everywhere.
Initially, it was implemented like that I just make it work for both feature because the eth feature wasn't building.